hw2/04-background.js

```diff
@@ -0,0 +1,51 @@
+// Add your code here
+const background = document.querySelector('.background');
+const startStop = document.querySelector('.start-stop');
+const intervalBox = document.querySelector('.interval-box');
+let currentDelay = 1000;
+let intervalID = null;
+
+background.style.background = 'rgb(25, 0, 255, 0.5)';
+
+const colorGen = function generateRandomColorValue() {
+  return Math.floor(Math.random() * 256);
+};
+
+const changeBackground = function changeBackgroundColor() {
+  background.style.background = `rgb(${colorGen()}, ${colorGen()}, ${colorGen()}, 0.5`;
+};
+
+const getInterval = function getCurrentIntervalInput(input) {
+  currentDelay = input.target.value * 1000;
+};
+
+const toggleButton = function toggleStartStopButton() {
+  if (startStop.textContent === 'Start') {
+    startStop.className = startStop.className.replace(
+      'btn-primary',
+      'btn-danger',
+    );
+    startStop.textContent = 'Stop';
+  } else {
+    startStop.className = startStop.className.replace(
+      'btn-danger',
+      'btn-primary',
+    );
+    startStop.textContent = 'Start';
+  }
```

Comment:
I like how you used `replace` to switch the button styles! One suggestion I have for this is to extract out the logic that styles the button into a function. As it stands, it looks like you repeat the same button style logic twice which violates the DRY principle. It's definitely not a big deal in the context of this homework though, but if you were making a larger web app having that function would make the program more maintainable.